docs/source/tox_environments.rst

```diff
@@ -0,0 +1,97 @@
+virtual environment and use this to install ``tox`` as a command line tool.  To
+install ``tox``, execute some version of
+
+.. code-block:: console
+
+    $ cd
+    $ deactivate (if already in a virtual environment)
+    $ /path/to/target/python --version
+    $ /path/to/target/python -m venv ~/.toxbase
+    $ ~/.toxbase/bin/pip list
+    $ ~/.toxbase/bin/python -m pip install --upgrade pip
+    $ ~/.toxbase/bin/python -m pip install --upgrade setuptools
+    $ ~/.toxbase/bin/python -m pip install tox
+    $ ~/.toxbase/bin/tox --version
+    $ ~/.toxbase/bin/pip list
+
+Rather than require that developers activate ``.toxbase``, we follow the
+suggestions in the `webinar <https://www.youtube.com/watch?v=PrAyvH-tm8E>`_
+and add ``tox`` to the ``PATH`` by executing an appropriate variation of
+
+.. code-block:: console
+
+    $ mkdir ~/local/bin
+    $ ln -s ~/.toxbase/bin/tox ~/local/bin/tox
+    $ vi ~/.bash_profile (add ~/local/bin to PATH)
+    $ . ~/.bash_profile
+    $ which tox
+    $ tox --version
+
+If the environment variable ``COVERAGE_FILE`` is set, then this is the name of
+the coverage results file that will be used with all coverage tasks;
+``COVERAGE_XML``, the name of the XML-format coverage report to generate;
+``COVERAGE_HTML``, the name of the HTML-format coverage report to generate.
+
+``tox`` will not carry out any work by default with the calls ``tox`` or ``tox
+-r``.  The following commands can be run from the root of a Taweret clone
+
+* ``tox -r -e coverage``
+
+  * Run the full Taweret test suite and save the coverage results to the
+    coverage file.
+  * Note that this task installs the Taweret code in its virtual environment as
+    an editable installation of the local clone, which mimics an interactive
+    developer setup.
+
+* ``tox -r -e nocoverage``
+
+  * Run the full Taweret test suite without coverage.
+  * Note that this task installs the Taweret code in its virtual environment by
+    building and installing a wheel, which mimics a user setup.
+
+* ``tox -r -e report``
+
+  * Generate XML- and HTML-format coverage reports based on the results of the
+    last execution of the ``coverage`` task.  Typically this will be run just
+    after or with ``coverage``
+
+* ``tox -r -e check``
+
+  * Run all Python code in the package through ``flake8`` to identify issues
+    and nonadherence to PEP8 standards.
+  * Note that this does **not** alter any files, but rather only reports issues.
+
+* ``tox -r -e html``
+
+  * Generate and render the Taweret documentation in HTML.
+
+Note that you can combine different tasks into a single call such as ``tox -e
+report,coverage``.
+
+The virtual environments created by ``tox`` can be activated for general use by
+developers.  In particular, the ``coverage`` environment can be useful for
+interactive development and testing since Taweret is installed in
+editable/developer mode.  Assuming that the ``coverage`` task has already been
```

Comment:
Might be useful to include a link to `setuptools` documentation on what _editable_ mode means?
https://setuptools.pypa.io/en/latest/userguide/development_mode.html

Reply:
Done.